### PR TITLE
add CAN safety button

### PR DESCRIPTION
**Pull request by CUAVcaijie**

### CODE REVIEW

**Comment by LorenzMeier** [comment]
@dagar Could you please provide quick feedback here?

File: src/drivers/uavcan/dsdl/com/hex/equipment/indication/20001.ButtonCommand.uavcan [hunk 1]
```diff
@@ -0,0 +1,6 @@
+#
+# Nodes that are capable of producing sounds should obey.
```
Comment (LorenzMeier) [inline src/drivers/uavcan/dsdl/com/hex/equipment/indication/20001.ButtonCommand.uavcan]: This is in a manufacturer namespace (Hex). It sounds like this should be in a standard namespace instead and be vendor-neutral.

Comment (CUAVcaijie) [reply src/drivers/uavcan/dsdl/com/hex/equipment/indication/20001.ButtonCommand.uavcan]: This part can be deleted

**Comment by dagar** [comment]
I'm not sure about the px4io interaction. In general any time a module/driver both subscribes and publishes the same topic it's a bit of concern and usually a sign we need to rethink things higher level.

Let me give this PR a try.

**Comment by hamishwillee** [comment]
@mrpollo This one seems a little stuck.

**Comment by CUAVcaijie** [comment]
Recently testing ICM20649 driver


------------------------------------------------------------------
发件人：Hamish Willee <notifications@github.com>
发送时间：2020年4月7日(星期二) 12:21
收件人：PX4/Firmware <Firmware@noreply.github.com>
抄　送：梁财杰 <caijie@cuav.net>; Author <author@noreply.github.com>
主　题：Re: [PX4/Firmware] add CAN safety button (#14319)

@mrpollo This one seems a little stuck.

**Comment by AlexandreBorowczyk** [comment]
Quick comment: we should add a check that the button have been depressed before switching state.
Such as
`[...]
	if (!_depressed && msg.press_time < 10) {
		_depressed = true;
	}

	if (msg.press_time > 10 && msg.button == 1 && _depressed) {
[...]`

**Pull request closed without merging**